client/state/notices/account-recovery/index.js

```diff
@@ -6,9 +6,43 @@ import { translate } from 'i18n-calypso';
 /**
  * Internal dependencies
  */
-import { errorNotice } from 'state/notices/actions';
+import {
+	successNotice,
+	errorNotice,
+} from 'state/notices/actions';
+
+import {
+	getAccountRecoveryPhone,
+	getAccountRecoveryEmail,
+} from 'state/account-recovery/settings/selectors';
+
 import { dispatchError } from '../utils';
 
+const getUpdateSuccessMessage = ( target, getState ) => {
+	switch ( target ) {
+		case 'phone':
+			const oldPhone = getAccountRecoveryPhone( getState() );
+
+			if ( null == oldPhone ) {
+				return translate( 'Successfully added new recovery SMS number.' );
+			}
+
+			return translate( 'Successfully updated recovery SMS number.' );
+
+		case 'email':
+			const oldEmail = getAccountRecoveryEmail( getState() );
+
+			if ( '' === oldEmail ) {
```

Comment:
Rather than `'' ===` or `null ===` would `! oldEmail` not be more fitting?  
It would more generically check all falsey values (including `''` and `null`) and so cares less about `getAccountRecoveryEmail`s implementation.